tests/Rule/Image/ImageTest.php

```diff
@@ -107,6 +199,56 @@ public function dataValidationFailed(): array
         ];
     }
 
+    public function dataValidationFailedAspectRatio(): array
+    {
+        return [
+            'default aspect ratio margin , smaller height' => [
+                new ImageInfo(800, 599),
+                new Image(aspectRatioWidth: 4, aspectRatioHeight: 3),
+                ['' => ['The aspect ratio of image "" must be 4:3 with margin 0%.']],
```

Comment:
Should it error with `""`? Isn't it better to omit it in the message in this case?

Reply:
It's already handled in #644, but the changes are in the branch for major release.